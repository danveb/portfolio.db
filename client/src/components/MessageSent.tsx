import { Button } from "./index"; 
import "../styles/MessageSent.css"; 

export default function MessageSent() {
  return (
    <div className="messageSent">
      <div className="messageSent__wrapper">
        <section className="messageSent__main">
          <div className="messageSent__title">
            <h1>thanks a lot<span className="dot--green">.</span> ðŸŽŠ</h1>
          </div>
          <div className="breakline"></div>
          <div className="messageSent__intro">
            <p>message sent | thank you for visiting my portfolio. And talk soon! </p>
          </div>
        </section>
        <section className="messageSent__bottom">
          <Button title="chau chau ðŸ‘‹" url="/" />
        </section>
      </div>
    </div>
  )
}