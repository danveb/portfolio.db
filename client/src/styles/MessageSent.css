.messageSent__wrapper {
  width: 100%; 
  height: 100%; 
}

.messageSent__main {
  display: flex; 
  flex-direction: column; 
  justify-content: center; 
  gap: 20px; 
  margin: 25px 0;
}

.messageSent__title {
  font-family: "Mulish", sans-serif; 
  font-size: 20px; 
  color: var(--alice-blue); 
  letter-spacing: 0.5px; 
}

.messageSent__intro {
  font-family: inherit; 
  font-size: 16px; 
  font-weight: 300; 
  color: #d4d4d4; 
  letter-spacing: 1px; 
  line-height: 1.5rem; 
}

.messageSent__bottom {
  margin-bottom: 40px; 
}

@media screen and (max-width: 1024px) {
  .messageSent__wrapper {
    padding: 0 20px; 
  }
}

@media screen and (max-width: 768px) {
  .messageSent__title {
    font-size: 16px; 
  }

  .messageSent__intro {
    font-size: 14px; 
  }
}