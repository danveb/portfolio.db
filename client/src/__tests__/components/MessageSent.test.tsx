import { render, screen } from "@testing-library/react"; 
import MessageSent from "../../components/MessageSent"; 
import { BrowserRouter } from "react-router-dom";

describe("MessageSent component", () => {
  // smoke test
  test("renders component wtesthout crashing", () => {
    render(
      <BrowserRouter>
        <MessageSent />
      </BrowserRouter>
    ); 
  });

  // snapshot test 
  test("matches snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <MessageSent />
      </BrowserRouter>
    ); 
    expect(asFragment()).toMatchSnapshot(); 
  }); 

  // getByText
  test("heading tag is displayed correctly", () => {
    render(
      <BrowserRouter>
        <MessageSent />
      </BrowserRouter>
    );
    const heading1 = screen.getByText("thanks a lot", {
      exact: false
    });
    expect(heading1).toBeInTheDocument(); 
  });
})