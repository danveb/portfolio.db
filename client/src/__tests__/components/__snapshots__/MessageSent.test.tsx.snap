// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageSent component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="messageSent"
  >
    <div
      class="messageSent__wrapper"
    >
      <section
        class="messageSent__main"
      >
        <div
          class="messageSent__title"
        >
          <h1>
            thanks a lot
            <span
              class="dot--green"
            >
              .
            </span>
             ðŸŽŠ
          </h1>
        </div>
        <div
          class="breakline"
        />
        <div
          class="messageSent__intro"
        >
          <p>
            message sent | thank you for visiting my portfolio. And talk soon! 
          </p>
        </div>
      </section>
      <section
        class="messageSent__bottom"
      >
        <div
          class="button__container"
          data-testid="button"
        >
          <a
            class="button--cta"
            href="/"
          >
            chau chau ðŸ‘‹
            <img
              alt="right arrow"
              src="right-arrow.png"
            />
          </a>
        </div>
      </section>
    </div>
  </div>
</DocumentFragment>
`;
